packages/contracts/src/contracts/webb-anchor.ts

```diff
@@ -196,56 +215,14 @@ export class WebbAnchorContract {
     return tree.path(leafIndex);
   }
 
-  async generateMerkleProofForRoot(deposit: Deposit, targetRoot: string, contractForCache: WebbAnchorContract) {
-    const bridgeStorageStorage = await bridgeCurrencyBridgeStorageFactory();
-    const bridgeStorageContract = await bridgeStorageStorage.get(this._contract.address);
-    logger.trace(`storage of leaves for contract ${this._contract.address}: `, bridgeStorageContract);
-    const storedContractInfo: MixerStorage[0] = (await bridgeStorageStorage.get(this._contract.address)) || {
-      lastQueriedBlock: anchorsStorage[this._contract.address.toString()] || 0,
-      leaves: [] as string[],
-    };
-
-    // const otheStoredContractInfo: MixerStorage[0] = (await bridgeStorageStorage.get(
-    //   contractForCache._contract.address
-    // )) || {
-    //   lastQueriedBlock: anchorsStorage[contractForCache._contract.address.toString()] || 0,
-    //   leaves: [] as string[],
-    // };
-    const treeHeight = await this._contract.levels();
-    logger.trace(`Generating merkle proof treeHeight ${treeHeight} of deposit`, deposit);
-    const tree = new MerkleTree('eth', treeHeight, [], new PoseidonHasher());
-    const lastQueriedBlock = storedContractInfo.lastQueriedBlock;
-
-    logger.trace('generate merkle proof with deposit', deposit, ` for the target roof ${targetRoot}`);
-
-    // Then, check the new leaves
-    const fetchedLeaves = await this.getDepositLeaves(lastQueriedBlock + 1, 0);
+  async generateLinkedMerkleProof(sourceDeposit: Deposit, sourceLeaves: string[]) {
+    // Get the source latest root
+    const latestSourceRoot = (await this._contract.getLatestNeighborRoots())[0];
```

Comment:
A more sustainable long term way to do this is to read the edges first. Then find the edge index of the source chain. Then grab the neighbor root at that index. Thoughts?